src/libraries/System.Data.OleDb/src/OleDbCommand.cs

```diff
@@ -400,7 +400,14 @@ private void CreateAccessor()
 
             bindings.AllocateForAccessor(null, 0, 0);
 
-            ApplyParameterBindings(commandWithParameters, bindings.BindInfo);
+            if (bindings.BindInfo_x86 != null)
+            {
+                ApplyParameterBindings(commandWithParameters, bindings.BindInfo_x86);
```

Comment:
You can make the `ApplyParameterBindings` to take `bindings` as an argument, and then do the switch for x86 inside the method. I believe there will be less duplicated code that way.